[Chung Yunseong] iP

JasonC01 opened the pull request:
## DukeTheGenious
> “Your mind is for having ideas, not holding them.”
> -David Allen

Never miss any of your schedule with DukeTheGenious. We support works like:
* Todos
* Jobs with deadlines
* Events happening at a specific time

Now, all you have to do to enjoy our fabulous app is,
1. Download it with just a click [here](https://github.com/JasonC01/ip)
2. Open the app 
3. Add your first task and,
4. You are ready to roll!😉

Major features:
- [x] Add and delete tasks
- [x] Mark your jobs done when finished.
- [x] Notifications (Future updates).   



Most importantly, all these amazing features come for **FREE**

Note our main function if you are a _JAVA_ programmer.
```java
public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
```

Commit:
Add Gradle support

JasonC01 committed:
Level-1

JasonC01 committed:
Level-2

JasonC01 committed:
Level-3

JasonC01 committed:
Level-4

JasonC01 committed:
Level-5

JasonC01 committed:
Level-6

JasonC01 committed:
A-TextUiTesting

JasonC01 committed:
Add saving feature

JasonC01 committed:
Add Date and Time features

JasonC01 committed:
Level-7 & Level-8

JasonC01 committed:
Add more oop

JasonC01 committed:
Add package

JasonC01 committed:
Add JUnit tests

JasonC01 committed:
Add javadoc

JasonC01 committed:
Follow coding standards

JasonC01 committed:
Add find function

JasonC01 committed:
Follow coding standards

JasonC01 committed:
Merge branch-A-JavaDoc

JasonC01 committed:
Merge branch-A-CodingStandard

JasonC01 committed:
Merge branch-Level-9

limzk126 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,26 @@
+        taskList.remove(taskIndex);
+    }
+    public int length() {
+        return taskList.size();
```
Would changing the method name to getLength() instead be better?

limzk126 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,26 @@
+        });
+    }
+    public Task get(int taskIndex) {
+        return taskList.get(taskIndex);
```
Would changing the method name to getTask() instead be better?

limzk126 reviewed (commented):
Good naming conventions, probably just need to fix some nits.

tianyue58 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    public Event(String taskContent) {
+        super(taskContent.split(" /at ")[0], "E");
+        eventTime = Parser.parseTiming(taskContent.split(" /at ")[1]);
```
Maybe Parser.parseTiming(taskContent.split(" /at ") could be an array variable. Would that make the code more readable? 🤔

tianyue58 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,35 @@
+    }
+    public static String parseTiming(String timing) {
+        String formattedDateTime = "";
+        String date= timing.split(" ")[0];
```
Should there be a spacing before "="? 🤔

tianyue58 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,30 @@
+import java.time.format.DateTimeFormatter;
+
+public class Task {
+    private final String taskContent;
```
Should name of final variables be capitalised? 🤔

tianyue58 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,108 @@
+    public String readLine() {
+        return scanner.nextLine();
+    }
+    public void handleInput(String input) {
```
Would it be better to extract some logic in this huge function to smaller functions? Maybe it would be more readable and easy to understand by others. :)

tianyue58 reviewed (commented):
LGTM. Just some small things to fix. Good job!

JasonC01 committed:
Add gradle

limzk126 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,136 @@
+            result += "|" + task.getTiming();
+        }
+        return result;
+    }
```
Consider using String.format() here, I think it enhances readability and saves the concatenation.

JasonC01 committed:
Add gradle

JasonC01 committed:
Add GUI

JasonC01 committed:
Add gradle

JasonC01 committed:
Add assertions in different parts of the code

Assertions are required to check for any possible bugs

Assertions will help us to validate whether the program is
functioning properly by checking our assumptions about
the code at various locations of the code.

Let’s add assertions in suspicious parts of the code where
the result of the code doesn’t turn out the way we expect it to be

Using assertions is preferable in the sense that it might help up to
catch unexpected bugs in our program.

JasonC01 committed:
Improve code quality

JasonC01 committed:
Merge pull request #1 from JasonC01/branch-A-Assertions

Add assertions in different parts of the code

JasonC01 committed:
Merge pull request #2 from JasonC01/branch-A-CodeQuality

Improve code quality

JasonC01 committed:
Add extension C-Undo

JasonC01 committed:
Merge pull request #3 from JasonC01/branch-BCD-Extension

Add extension C-Undo

JasonC01 committed:
Use enums

JasonC01 committed:
Add readme

JasonC01 committed:
Set theme jekyll-theme-cayman

JasonC01 committed:
Add Ui.png

JasonC01 committed:
Add Ui.png

JasonC01 committed:
Fix minor bugs

JasonC01 committed:
Create jar file for javaFX

JasonC01 committed:
Fix final bugs

JasonC01 committed:
Fix readme

JasonC01 committed:
Fix readme
